src/android.rs

```diff
@@ -105,7 +150,101 @@ pub mod android {
         let bucket_id = jstring_to_string(&env, java_bucket_id);
         match openDatastore().get_events(&bucket_id, None, None, None) {
             Ok(events) => string_to_jstring(&env, json!(events).to_string()),
-            Err(_) => string_to_jstring(&env, "Something went wrong when trying to send heartbeat".to_string())
+            Err(_) => create_error_object(&env, "Something went wrong when trying to send heartbeat".to_string())
+        }
+    }
+
+    // From: https://github.com/servo/servo/pull/21812/files
+    // With modifications from latest: https://github.com/servo/servo/blob/5de6d87c97050db35cfb0a575e14b4d9b6207ac5/ports/libsimpleservo/jniapi/src/lib.rs#L480
+    use std::os::raw::{c_char, c_int};
+    use std::thread;
+
+    extern "C" {
+        pub fn __android_log_write(prio: c_int, tag: *const c_char, text: *const c_char) -> c_int;
+    }
+
+    fn redirect_stdout_to_logcat() {
```

Comment:
Consider splitting android.rs into a android folder and keep this copy-pasted utilities code in a separate source file.